components/camera/transformpipeline/mods.go

```diff
@@ -138,3 +138,68 @@ func (rs *resizeSource) Read(ctx context.Context) (image.Image, func(), error) {
 func (rs *resizeSource) Close(ctx context.Context) error {
 	return rs.originalStream.Close(ctx)
 }
+
+// cropConfig are the attributes for a crop transform.
+type cropConfig struct {
+	XMin int `json:"x_min_px"`
+	YMin int `json:"y_min_px"`
+	XMax int `json:"x_max_px"`
+	YMax int `json:"y_max_px"`
+}
+
+type cropSource struct {
+	originalStream gostream.VideoStream
+	imgType        camera.ImageType
+	cropWindow     image.Rectangle
+}
+
+// newCropTransform creates a new crop transform.
+func newCropTransform(
+	ctx context.Context, source gostream.VideoSource, stream camera.ImageType, am utils.AttributeMap,
+) (gostream.VideoSource, camera.ImageType, error) {
+	conf, err := resource.TransformAttributeMap[*cropConfig](am)
+	if err != nil {
+		return nil, camera.UnspecifiedStream, err
+	}
+	if conf.XMin >= conf.XMax {
+		return nil, camera.UnspecifiedStream, errors.New("cannot crop image to 0 width (x_min is >= x_max)")
+	}
+	if conf.YMin >= conf.YMax {
+		return nil, camera.UnspecifiedStream, errors.New("cannot crop image to 0 height (y_min is >= y_max)")
+	}
+	cropRect := image.Rect(conf.XMin, conf.YMin, conf.XMax, conf.YMax)
+
+	reader := &cropSource{gostream.NewEmbeddedVideoStream(source), stream, cropRect}
+	src, err := camera.NewVideoSourceFromReader(ctx, reader, nil, stream)
+	if err != nil {
+		return nil, camera.UnspecifiedStream, err
```

Comment:
Have we considered making the expected behavior for a bad crop return the original video source (as opposed to nothing)? Not particularly opinionated here just thinking about it.

Reply:
Best to error if there is an error, especially if it's essentially a config problem. Crops which will return a 0 image won't allow the transform component to be built.